android/filament-android/src/main/java/com/google/android/filament/View.java

```diff
@@ -582,6 +583,46 @@
         VSM
     }
 
+    /**
+     * View-level options for Shadowing.
+     *
+     * @see View#setShadowOptions
+     */
+    public static class ShadowOptions {
+        /**
+         * The shadow mapping technique this View uses.
+         *
+         * The ShadowType affects all the shadows seen within the View.
+         *
+         * <p>
+         * {@link ShadowType#VSM} imposes a restriction on marking renderables as only shadow
+         * receivers (but not casters). To ensure correct shadowing with VSM, all shadow participant
+         * renderables should be marked as both receivers and casters. Objects that are guaranteed
+         * to not cast shadows on themselves or other objects (such as flat ground planes) can be
+         * set to not cast shadows, which might improve shadow quality.
+         * </p>
+         *
+         * <strong>Warning: This API is still experimental and subject to change.</strong>
+         */
+        public ShadowType shadowType;
```

Comment:
You should set the default value of this field to PCF, it will be null otherwise. Also maybe mark this field as @NonNull